web/lib/api/db/s3Client.ts

```diff
@@ -0,0 +1,76 @@
+import { S3Client as AwsS3Client, GetObjectCommand } from "@aws-sdk/client-s3";
+import { getSignedUrl } from "@aws-sdk/s3-request-presigner";
+import { Result } from "../../result";
+
+const S3_ACCESS_KEY = process.env.S3_ACCESS_KEY ?? "";
+const S3_SECRET_KEY = process.env.S3_SECRET_KEY ?? "";
+const S3_ENDPOINT = process.env.S3_ENDPOINT ?? "";
+const S3_BUCKET_NAME = process.env.S3_BUCKET_NAME ?? "";
+
+// S3 endpoint can be empty string
+if (
+  !S3_ACCESS_KEY ||
+  !S3_SECRET_KEY ||
+  S3_ENDPOINT === null ||
+  S3_ENDPOINT === undefined ||
+  !S3_BUCKET_NAME
+) {
+  throw new Error("S3 env variables not set");
+}
+
+export type RequestResponseBody = {
+  request?: any;
+  response?: any;
+};
+
+export class S3Client {
+  private region = "us-west-2";
+  private awsClient: AwsS3Client;
+
+  constructor(
+    accessKey: string,
+    secretKey: string,
+    endpoint: string,
```

Comment:
Why does this need to be passed in if they are the same global variable every time?

Reply:
Either way can work. Passing in does not couple it to the environment variables allowing you to specify on initialization.